File: 3208. Alternating Groups II.py
class Solution(object):
    def numberOfAlternatingGroups(self, nums, k):
        n = len(nums)
        i, j, cnt, ans = 0, 0, 0, 0
        nums += nums

        while j < (n+k)-1:
            if nums[j] == nums[j+1]: i = j
            j += 1
            if j-i >= k: ans += 1

        return ans

class Solution:
    def numberOfAlternatingGroups(self, colors: List[int], k: int) -> int:
        i, j = 0, 0
        max_alternative = 0
        ans, n = 0, len(colors)
        colors += colors

        while(j < n + k - 1):
            if j and colors[j] == colors[j - 1]:
                max_alternative = 0
            max_alternative += 1

            if j - i + 1 == k:
                i += 1
                ans += max_alternative >= k
            j += 1
        return ans

File: 3208. Alternating Groups II.cpp
class Solution {
    public:
        int numberOfAlternatingGroups(vector<int>& nums, int k) 
        {
            int cnt = 0, ans = 0;
            int i = 0, j = 0, n = nums.size();
            for(int i=0;i<k;i++)
                nums.push_back(nums[i]);
    
            while(j<(n+k)-1)
            {
                if(nums[j++] == nums[j]) i = j-1;
                if(j-i >= k) ans++;
            }
            return ans;
        }
    };